driver/config/.schema/config.schema.json

```diff
@@ -403,34 +428,101 @@
         "id",
         "provider",
         "client_id",
-        "client_secret",
```

Comment:
`client_secret` is not required for Apple. I'm adding is as required in the condition below: https://github.com/ory/kratos/pull/1782/files#diff-c0162691c6563183b7fde70a705098b8460397b751f4176f671733ed40d3b1aeR489

Reply:
This gets a bit messy now :sweat_smile: I will restructure it at some point.... tracked as #1783

Reply:
Oh, that's very much true. Especially with both Apple and Microsoft having custom fields and `else` sections in the condition.
I'm wondering how strict we want to be when validating especially with fields like `tenant` or `private_key` being forbidden for providers that do not use of them. It makes the schema validation more complicated. And using `tenant` with Github is quite similar to using `issuer_url` with Facebook or Google as those providers override it anyway.